Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,355 @@
+            {
+                if (i == peopleAndBalances.Length - 1)
+                {
+                    RichestPerson.Append(" and ");
+                }
+                else
+                {
+                    RichestPerson.Append(", ");
+                }
+
+                RichestPerson.Append(currentPersonData[0]);
+            }
+        }
+
+        private static decimal FindRichestPerson(decimal HighestBalance, StringBuilder RichestPerson, string[] currentPersonData, decimal highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson > HighestBalance)
+            {
+                HighestBalance = highestAmountOfPerson;
+                var currentOne = RichestPerson.ToString();
+
+                if (string.IsNullOrEmpty(currentOne))
+                {
+                    RichestPerson.Append(currentPersonData[0]);
+                }
+                else
+                {
+                    RichestPerson.Clear();
+                    RichestPerson.Append(currentPersonData[0]);
+                }
+
+
+            }
+
+            return HighestBalance;
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var lowestBalance = int.MaxValue;
+            var poorestPerson = new StringBuilder();
+
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var isValid = int.TryParse(currentPersonData[currentPersonData.Length - 1], out int highestAmountOfPerson);
+
+                if (!isValid)
+                {
+                    return "N/A.";
+                }
+
+                lowestBalance = FindPoorestPerson(peopleAndBalances, lowestBalance, poorestPerson, i, currentPersonData, highestAmountOfPerson);
+
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            NumberFormatInfo noParens = (NumberFormatInfo)CultureInfo.CurrentCulture.NumberFormat.Clone();
+            noParens.CurrencyNegativePattern = 1; 
+            var lowestBalanceString = lowestBalance.ToString("C0", noParens);
+
+            var answer = $"{poorestPerson} has the least money. {lowestBalanceString}.";
+
+            TextTable.Build(answer, 3);
+
+            return answer;
+
+
+        }
+
+        private static int FindPoorestPerson(string[] peopleAndBalances, int lowestBalance, StringBuilder poorestPerson, int i, string[] currentPersonData, int highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson < lowestBalance)
+            {
+                lowestBalance = highestAmountOfPerson;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'lowestBalance'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)

Reply:
Looks legal to me so I do not understand why this is wrong